### DISCUSSION TITLE
Dynamic Helm values for multi-environment scenarios

### DISCUSSION THREAD

**Discussion by mikecameronh1** [kcl-lang/kcl#1774] [Q&A] [answered]
We rely on a lot of OSS Helm charts and would like to gauge the feasibility of using KCL and the Helm KCL plugin to hydrate Helm charts using dynamic values (`values.yaml`). Right now, we are using [Tanka](https://tanka.dev/), which has the ability to do just this using https://tanka.dev/helm/; basically allowing you to template a Helm chart with arbitrary values that we configure for each environment. This also enables us to patch once the Helm chart has been template.

I should add that I know this could be achieved using a few commands:

1. generate `values.yaml` from KCL
2. feed generated values to `helm template`

Just wondering if there's a more integrated way of doing so.

What I'm having trouble with is understanding how to achieve this with KCL and `KCLRun`.

**Comment by Peefy** [comment]
cc @He1pa @zong-zhe

**Comment by mikecameronh1** [comment, marked as answer]
By sheer luck, I found https://github.com/cakehappens/kcfoil/tree/main. Which funny enough gave me exactly what I was looking for.

Thanks again.
